src/Analyzers/CSharp/Tests/RemoveUnnecessaryCast/RemoveUnnecessaryCastTests.cs

```diff
@@ -7228,34 +7738,32 @@ static void Main()
 
         [WorkItem(20211, "https://github.com/dotnet/roslyn/issues/20211")]
         [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsRemoveUnnecessaryCast)]
-        public async Task RemoveDoubleNullCastInSwitch1()
+        public async Task DoNotRemoveDoubleNullCastInSwitch1()
```

Comment:
this was a bug before.  the cast should not be removable as it can cause reachability errors.  the new test demonstrates the case for this.